Build failure: digimend drivers failing to compile with kernel 6.3.x

### Steps To Reproduce

```
services.xserver.digimend.enable = true;
boot.kernelPackages = pkgs.linuxPackages_6_3;
```

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/10saww0ib3kff0k4zj5bafh9mxg07x8d-source
source root is source
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "configurePhase" }
configuring
no configure script, doing nothing
@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: SHELL=/nix/store/j80wcbm6bb1nsqsgmniq43rg980p27q9-bash-5.2-p15/bin/bash O=\$\(buildRoot\) CC=/nix/store/b5mkki8bysygaf7a01jpj88zrxibkrv4-gcc-wrapper-12.2.0/bin/cc HOSTCC=/nix/store/b5mkki8bysygaf7a01jpj88zrxibkrv4-gcc-wrapper-12.2.0/bin/cc HOSTLD=/nix/store/0kh9lkqqjpwflxfvb4a1ngxnkcfnlspd-binutils-wrapper-2.40/bin/ld ARCH=x86_64 KVERSION=6.3.1 KDIR=/nix/store/q6m4p67s6l9dh9wall03fkjbylrmpxsp-linux-6.3.1-dev/lib/modules/6.3.1/build DESTDIR=/nix/store/vnwg0jmxysc4a0r153l9pkjswgx5zf33-digimend-10 INSTALL_MOD_PATH=/nix/store/vnwg0jmxysc4a0r153l9pkjswgx5zf33-digimend-10
make -C /nix/store/q6m4p67s6l9dh9wall03fkjbylrmpxsp-linux-6.3.1-dev/lib/modules/6.3.1/build M=/build/source modules
  CC [M]  /build/source/hid-kye.o
  CC [M]  /build/source/hid-uclogic-core.o
  CC [M]  /build/source/hid-uclogic-rdesc.o
  CC [M]  /build/source/hid-uclogic-params.o
/build/source/hid-uclogic-params.c: In function 'uclogic_params_init':
/build/source/hid-uclogic-params.c:1030:17: error: implicit declaration of function 'hid_is_using_ll_driver' [-Werror=implicit-function-declaration]
 1030 |             || !hid_is_using_ll_driver(hdev, &usb_hid_driver)
      |                 ^~~~~~~~~~~~~~~~~~~~~~
/build/source/hid-uclogic-params.c:1030:47: error: 'usb_hid_driver' undeclared (first use in this function); did you mean 'to_hid_driver'?
 1030 |             || !hid_is_using_ll_driver(hdev, &usb_hid_driver)
      |                                               ^~~~~~~~~~~~~~
      |                                               to_hid_driver
/build/source/hid-uclogic-params.c:1030:47: note: each undeclared identifier is reported only once for each function it appears in
/build/source/hid-uclogic-params.c:1147:20: warning: this statement may fall through [-Wimplicit-fallthrough=]
 1147 |                 if (bNumInterfaces != 3) {
      |                    ^
/build/source/hid-uclogic-params.c:1165:9: note: here
 1165 |         case VID_PID(USB_VENDOR_ID_HUION,
      |         ^~~~
cc1: some warnings being treated as errors
make[2]: *** [/nix/store/q6m4p67s6l9dh9wall03fkjbylrmpxsp-linux-6.3.1-dev/lib/modules/6.3.1/source/scripts/Makefile.build:252: /build/source/hid-uclogic-params.o] Error 1
make[1]: *** [/nix/store/q6m4p67s6l9dh9wall03fkjbylrmpxsp-linux-6.3.1-dev/lib/modules/6.3.1/source/Makefile:2025: /build/source] Error 2
make: *** [Makefile:25: modules] Error 2
```

### Additional context

This looks similar: https://bugs.launchpad.net/ubuntu/+source/digimend-dkms/+bug/2011546 according to which the solution is getting the kernel built with `CONFIG_USB=y` or `CONFIG_USB=m`.
Edit: just noticed that while the error looks similar, it's for kernel 5.15 so probably a different issue.

### Notify maintainers
@gebner 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/ivfz7lhgf8lw3q14dxxhf3xmbsa06r0z-nix-info
copying path '/nix/store/ivfz7lhgf8lw3q14dxxhf3xmbsa06r0z-nix-info' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.14, NixOS, 23.05 (Stoat), 23.05.20230506.897876e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos"`
 - channels(ivan): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

--- a/pkgs/os-specific/linux/digimend/default.nix
+++ b/pkgs/os-specific/linux/digimend/default.nix
@@ -2,13 +2,13 @@
 
 stdenv.mkDerivation rec {
   pname = "digimend";
-  version = "10";
+  version = "unstable-2023-05-03";
 
   src = fetchFromGitHub {
     owner = "digimend";
     repo = "digimend-kernel-drivers";
-    rev = "v${version}";
-    sha256 = "0lifd6cx6aa6hcms4zn4hlla3alra08r0svj5x1l8nlsv0ydnl6i";
+    rev = "eca6e1b701bffb80a293234a485ebf6b4bc85562";
+    hash = "sha256-0mjIUgHvbNcVQVzU3xzaloe5R41a4eknDhdhruJH+6c=";
   };
 
   postPatch = ''

